extensions/amp-social-share/0.2/social-share.js

```diff
@@ -30,63 +32,196 @@ const DEFAULT_HEIGHT = 44;
  * @param {!JsonObject} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare2(props) {
+  const name = 'SocialShare2';
+  useResourcesNotify();
+
+  const {
+    typeConfig,
+    baseEndpoint,
+    checkedWidth,
+    checkedHeight,
+    obsoleteType,
+  } = checkProps_(props, name);
+  if (obsoleteType) {
+    return;
+  }
+  const finalEndpoint = createEndpoint_(typeConfig, baseEndpoint, props);
+
   /**
-   * Handle key presses on the element.
-   * @param {!Event} event
    * @private
+   * @param {!JsonObject} props
+   * @param {?string} name
+   * @return {!JsonObject}
    */
-  function handleKeyPress(event) {
-    const {key} = event;
-    if (key == Keys.SPACE || key == Keys.ENTER) {
-      event.preventDefault();
-      handleActivation();
+  function checkProps_(props, name) {
+    const {type, shareEndpoint, params, bindings, width, height} = props;
+
+    // Verify type is valid and cannot contain spaces
+    if (type === undefined) {
+      throwError_(`The type attribute is required. ${name}`);
+    }
+    if (/\s/.test(type)) {
+      throwError_(
+        `Space characters are not allowed in type attribute value. ${name}`
+      );
+    }
+
+    // bindings and params props must be objects
+    if (params && !isObject(params)) {
+      throwError_(`The params property should be an object. ${name}`);
+    }
+    if (bindings && !isObject(bindings)) {
+      throwError_(`The bindings property should be an object. ${name}`);
+    }
+
+    // User must provide shareEndpoint if they choose a type that is not
+    // pre-configured
+    const typeConfig = getSocialConfig(type) || dict();
+    const baseEndpoint = typeConfig['shareEndpoint'] || shareEndpoint;
+    if (baseEndpoint === undefined) {
+      throwError_(
+        `A shareEndpoint is required if not using a pre-configured type. ${name}`
+      );
+    }
+
+    // Throw warning if type is obsolete
+    const obsoleteType = typeConfig['obsolete'];
+    if (obsoleteType) {
+      throwWarning_(`Skipping obsolete share button ${type}. ${name}`);
     }
+
+    // Verify width and height are valid integers
+    let checkedWidth = width || DEFAULT_WIDTH;
+    let checkedHeight = height || DEFAULT_HEIGHT;
+    if (width && !Number.isInteger(width)) {
```

Comment:
Non-integer `width` and `height` are generally totally ok, even if a bit uncommon.